digiwf-engine/digiwf-engine-service/src/main/java/de/muenchen/oss/digiwf/humantask/process/listener/UserTaskNotificationListener.java

```diff
@@ -62,10 +62,28 @@ public class UserTaskNotificationListener {
 
     @EventListener
     public void delegateTask(final DelegateTask delegateTask) throws Exception {
+        // send notification on task creation for candidate users and groups
         if (delegateTask.getEventName().equals("create")) {
-
             log.debug("Notification for created task: {}", delegateTask.getName());
+            val notifyCandidateUsers = NOTIFICATION_SEND_CANDIDATE_USERS.from(delegateTask).getOptional();
+            val notifyCandidateUsersV02 = ProcessTaskConstants.APP_NOTIFICATION_SEND_CANDIDATE_USERS.from(delegateTask).getOptional();
+            if ((!notifyCandidateUsers.isPresent() || "true".equals(notifyCandidateUsers.get()))
+                    && (!notifyCandidateUsersV02.isPresent() || "true".equals(notifyCandidateUsersV02.get()))) {
+                this.notifyCandidateUsers(delegateTask);
+            }
 
+            val notifyCandidateGroups = NOTIFICATION_SEND_CANDIDATE_GROUPS.from(delegateTask).getOptional();
+            val notifyCandidateGroupsV02 = ProcessTaskConstants.APP_NOTIFICATION_SEND_CANDIDATE_GROUPS.from(delegateTask).getOptional();
+            if ((!notifyCandidateGroups.isPresent() || "true".equals(notifyCandidateGroups.get()))
+                    && (!notifyCandidateGroupsV02.isPresent() || "true".equals(notifyCandidateGroupsV02.get()))) {
+                this.notifyCandidateGroups(delegateTask);
+            }
+        }
+        // send notification on task assignment for assignee
+        // assignment event is used for direct user assignment
+        // update event is used for group task assignment
+        else if (delegateTask.getEventName().equals("assignment")) {
```

Comment:
The assignment event is triggered directly after task creation if an assignee is specified. But it's also triggered on grouptask assignment. Therefore users can notify the group and the assignee which makes sense to me